docs/extensions/storage/proposedchanges.md

```diff
@@ -0,0 +1,185 @@
+
+TODO discuss to clarify further
+
+## Storage drivers
+
+### allow clients to send a uuid on upload
+iOS clients can only queue single requests to be executed in the background. They queue an upload and need to be able to identify the uploaded file after it has been uploaded to the server. The disconnected nature of the connection might cause workflows or manual user interaction with the file on the server to move the file to a different place or changing the content while the device is offline. However, on the device users might have marked the file as favorite or added it to other iOS specific collections. To be able to reliably identify the file the client can generate a `uuid` and attach it to the file metadata during the upload. While it is not necessary to look up files by this `uuid` having a second file id that serves exactly the same purpose as the `file id` is redundant.
+
+Another aspect for the `file id` / `uuid` is that it must be a logical identifier that can be set, at least by internal systems. Without a writeable fileid we cannot restore backups or migrate storage spaces from one storage provider to another storage provider.
+
+Technically, this means that every storage driver needs to have a map of a `uuid` to an internal resource identifier. This internal resource identifier can be
+- an eos fileid, because eos can look up files by id
+- an inode if the filesystem and the storage driver support looking up by inode
+- a path if the storage driver has no way of looking up files by id.
+  - In this case other mechanisms like inotify, kernel audit or a fuse overlay might be used to keep the paths up to date.
+  - to prevent excessive writes when deep folders are renamed a reverse map might be used: it will map the `uuid` to `<parentuuid>:<childname>`, in order to trade writes for reads
+  - as a fallback a sync job can read the file id from the metadata of the resources and populate the uuid to internal id map.
+
+The TUS upload can take metadata, for PUT we might need a header.
+
+### Space id vs resource id vs storage id
+
+We have `/dav/meta/<fileid>` where the `fileid` is a string that was returned by a PROPFIND or by the `/graph/v1.0/me/drives/` endpoint? That returns a space id and the root drive item which has an `id` 
+
+Does that `id` have a specific format? We currently concatenate as `<storageid>!<nodeid>`.
+
+A request against `/dav/meta/fileid` will use the reva storage registry to look up a path. 
+
+What if the storage space is moved to another storage provider. This happens during a migration:
```

Comment:
"During migrations, a storage space can be moved between storage providers. This is what happens:"